buildkitd/Earthfile

```diff
@@ -91,4 +91,12 @@ update-buildkit-earthfile:
     LOCALLY
     ARG --required BUILDKIT_GIT_SHA
     ARG --required BUILDKIT_GIT_ORG
-    RUN sed -i 's/\(^[ \t]\+ARG BUILDKIT_BASE_IMAGE=github.com\/'$BUILDKIT_GIT_ORG'\/buildkit\).*/\1:'$BUILDKIT_GIT_SHA'+build/g' Earthfile
+    # We run the Mac version on Darwin as it is not compatible with Linux sed
+    RUN if [ "$(uname)" = "Darwin" ]; then \
+            sed -E -i '' -e 's/(^[ \t]+ARG BUILDKIT_BASE_IMAGE).*\/buildkit.*/\1=github.com\/'$BUILDKIT_GIT_ORG'\/buildkit:'$BUILDKIT_GIT_SHA'+build/g' Earthfile; \
+        else \
+            "sed -i 's/\(^[ \t]\+ARG BUILDKIT_BASE_IMAGE\)=github.com\/.*\/buildkit.*/\1=github.com\/'$BUILDKIT_GIT_ORG'\/buildkit:'$BUILDKIT_GIT_SHA'+build/g' Earthfile"; \
```

Comment:
why is there a `"` before `sed`?

furthermore, this regex seems to be different compared with the old one -- is this intentional?

Reply:
Removed quotes.

The regex is intentionally different. Unfortunately, I was unable to get a regex that worked on both systems. They should be functionally equivalent though.

Reply:
I understand that the MacOS regex is different; however don't understand why the sed command for linux needs to change.

Reply:
Oh, it changed because we couldn't change the ORG using that version.
the old one was:
```
sed -i 's/\(^[ \t]\+ARG BUILDKIT_BASE_IMAGE=github.com\/'$BUILDKIT_GIT_ORG'\/buildkit\).*/\1:'$BUILDKIT_GIT_SHA'+build/g' Earthfile
```
By including `$BUILDKIT_GIT_ORG` it only matched if the organization was the same.

Reply:
oh sweet! that makes sense. nice change.